### ISSUE TITLE
puppet-lint doesn't handle CRLFs very well

### ISSUE THREAD

**Issue by glennsarti** [rodjek/puppet-lint#778]
In the puppet language server for VS Code we keep a copy of the manifest in memory as a string and then pass that into the puppet-lint validator.  This means I bypass the file loading that's present in https://github.com/rodjek/puppet-lint/blob/0799152d9ded067e1a3c5b2fa4f2748a30fbf2c5/lib/puppet-lint.rb#L78-L80

As the file loader is using text based loading (as opposed to binary), normally in puppet-lint any CRLFs are automatically converted to LFs puppet-lint doesn't have to deal with CRLFs

Repro:

Using a manifest of
``` puppet
exec { 'Wee!':
  command => 'echo The quick brown fox jumps over the lazy dog. The quick brown fox jumps over the lazy dog. The quick brown fox jumps over the lazy dog.', # lint:ignore:140chars
}

exec { 'Wee!':
  command => 'echo The quick brown fox jumps over the lazy dog. The quick brown fox jumps over the lazy dog. The quick brown fox jumps over the lazy dog.',
}

exec { 'Wee!':
  command  => 'echo The quick brown fox jumps over the lazy dog. The quick brown fox jumps over the lazy dog. The quick brown fox jumps over the lazy dog.', # lint:ignore:140chars
}

exec { 'Wee!':
  command => 'echo The quick brown fox jumps over the lazy dog. The quick brown fox jumps over the lazy dog. The quick brown fox jumps over the lazy dog.',
}
```

Saved with LFs or using puppet-lint gem "normally" generates two warnings:
```
WARNING: line has more than 140 characters on line 6
WARNING: line has more than 140 characters on line 14
```

Using CRLFs and binary mode, generates three warnings
```
WARNING: line has more than 140 characters on line 6
WARNING: line has more than 140 characters on line 10
WARNING: line has more than 140 characters on line 14
```

While the language server is using puppet-lint in a "not normal" way (directly in ruby instead of via CLI), I'm raising this issue as the number of issues seems odd.  Why did it honor the ignore on line 2, but not line 10.

I'm not expecting puppet-lint to support CRLFs (although if you could PLEASE!!!) but this may or may not be uncovering errors in the way warnings are ignored.

Ref - https://github.com/jpogran/puppet-vscode/issues/176

**Comment by rodjek** [comment]
Interesting, I thought I had supported CRLF everywhere where I'm parsing line endings, but I'll have to go back and look because I've clearly missed a place! :)

**Comment by glennsarti** [comment]
Do we have an ETA on releasing this commit?

**Comment by glennsarti** [comment]
@rnelson0 ^^
